crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -424,9 +422,37 @@ module sui::staking_pool {
             pool_token_amount: balance::supply_value(&pool.delegation_token_supply),
         }
     }
+
+    /// Calculate the total value of the pending staking requests for this staking pool.
+    public fun pending_stake_amount(staking_pool: &StakingPool): u64 {
+        let sum = 0;
+        let elem_opt = *linked_table::front(&staking_pool.pending_delegations);
+
+        while (option::is_some(&elem_opt)) {
+            let elem_key = *option::borrow(&elem_opt);
+            let pending_delegation_entry = linked_table::borrow(&staking_pool.pending_delegations, elem_key);
+            sum = sum + pending_delegation_entry.sui_amount;
+            elem_opt = *linked_table::next(&staking_pool.pending_delegations, elem_key);
+        };
+        sum
+    }
+
+    /// Calculate the current the total withdrawal requests (in terms of principal) for the staking pool
+    public fun pending_principal_withdrawal_amounts(staking_pool: &StakingPool): u64 {
```

Comment:
This and the function above walk through the (potentially pretty big) table every time and do a lot of dynamic child loading. Maybe keep a rolling sum in the struct instead?

Reply:
Good question! I believe (and will need to double-check) that these functions are only used in tests, so not sure if we should add a field to the structs to keep the rolling sum. If you think these will be useful to have in general as non-test helpers, happy to make this change :)

Reply:
Let's keep it simple and make it test only then :)

Reply:
Actually had a look and it appears as though this function is used by `frenemies` so I think we might need to add this field in unless we want to depricate that functionality over there 😢

Reply:
Seems like frenemies is getting a soft deprecation, so I vote we keep this as-is for now, and we can remove later if that sounds good?